src/rpm/headers/types.rs

```diff
@@ -32,30 +32,32 @@ pub struct PackageFileEntry {
     pub(crate) content: Vec<u8>,
 }
 
-#[non_exhaustive]
 #[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
-pub enum FileMode {
-    // It does not really matter if we use u16 or i16 since all we care about
-    // is the bit representation which is the same for both.
-    Dir { permissions: u16 },
-    Regular { permissions: u16 },
-    SymbolicLink { permissions: u16 },
-    // For "Invalid" we use a larger integer since it is possible to create an invalid
-    // FileMode by providing an overflowing integer.
-    Invalid { raw_mode: i32, reason: &'static str },
```

Comment:
I don't think this should be one of our "valid" states, we can just return an error.  If we want to parse invalid RPMs then we can compute the mode lazily.